src/engine/QueryPlanner.h

```diff
@@ -57,13 +58,13 @@ class QueryPlanner {
       Node(size_t id, SparqlTriple t,
            std::optional<Variable> graphVariable = std::nullopt)
           : id_(id), triple_(std::move(t)) {
-        if (isVariable(triple_.s_)) {
+        if (triple_.s_.isVariable()) {
           _variables.insert(triple_.s_.getVariable());
         }
-        if (isVariable(triple_.p_)) {
-          _variables.insert(Variable{triple_.p_.iri_});
+        if (std::holds_alternative<Variable>(triple_.p_)) {
+          _variables.insert(std::get<Variable>(triple_.p_));
```

Comment:
Make the interface symmetric (s..t you can also write `isVariable` and `getVariable`.